docs/articles/documentation/extending-testcafe/custom-browser-provider-plugin/index.md

```diff
@@ -0,0 +1,180 @@
+---
+layout: docs
+title: Custom Browser Provider Plugin
+permalink: /documentation/extending-testcafe/custom-browser-provider-plugin/
+---
+# Custom Browser Provider Plugin
+
+To work with different browsers during testing, TestCafe can use the [built-in library](../../using-testcafe/common-concepts/using-browsers.md#built-in-library) or [browser provider plugins](../../using-testcafe/common-concepts/using-browsers.md#browser-provider-plugins).
+If you need to use a specific browser and perform actions on the browser window in a custom way, you can create a **custom browser provider plugin**. For this purpose, use the [TestCafe browser provider generator](https://github.com/DevExpress/generator-testcafe-browser-provider).
+The generator will scaffold out the plugin, so that you only need to write a few lines of code.
+
+For more information on how to implement the plugin, see the following sections.
+
+* [Generating Browser Provider Project](#generating-browser-provider-project)
+* [Implementing the Browser Provider](#implementing-the-browser-provider)
+* [Building the Provider](#building-the-provider)
+* [Using the Provider Development Version](#using-the-provider-development-version)
+* [Publishing the Provider to NPM](#publishing-the-provider-to-npm)
+* [Reference](#reference)
+
+## Generating Browser Provider Project
+
+First, install [Yeoman](http://yeoman.io) and `generator-testcafe-browser-provider` using [npm](https://www.npmjs.com/).
+
+```bash
+npm install -g yo
+npm install -g generator-testcafe-browser-provider
+```
+
+Create a new directory where the generator should place your scaffolded project files and go into it.
+
+```bash
+mkdir my-browser-provider
+cd my-browser-provider
+```
+
+> It is recommended to name the directory as you would like to name the browser provider project. When you run the generator, it will automatically suggest the browser provider name that matches the browser provider directory name. As for the provider package name, it will consist of two parts - the `{testcafe-browser-provider-}` prefix and the name of the provider itself, for example, `{testcafe-browser-provider-my-provider}`.
```

Comment:
We need to mention that it's a requirement that plugin name should start from `testcafe-browser-provider-`, otherwise TestCafe will not be able to find it